Content.Server/GameObjects/Components/Interactable/ToolComponent.cs

```diff
@@ -0,0 +1,182 @@
+        {
+            get => _behavior;
+            set
+            {
+                _behavior = value;
+                Dirty();
+            }
+        }
+
+        /// <summary>
+        ///     For tool interactions that have a delay before action this will modify the rate, time to wait is divided by this value
+        /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float SpeedModifier
+        {
+            get => _speedModifier;
+            set => _speedModifier = value;
+        }
+
+        public string UseSound
+        {
+            get => _useSound;
+            set => _useSound = value;
+        }
+
+        public string UseSoundCollection
+        {
+            get => _useSoundCollection;
+            set => _useSoundCollection = value;
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            _audioSystem = _entitySystemManager.GetEntitySystem<AudioSystem>();
+            _interactionSystem = _entitySystemManager.GetEntitySystem<InteractionSystem>();
+            Owner.TryGetComponent(out _spriteComponent);
+        }
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+
+            if (serializer.Reading)
+            {
+                try
+                {
+                    _behavior = (Tool)serializer.ReadStringEnumKey("behavior");
+                }
+                catch
+                {
+                    // ignored
+                }
+            }
+            serializer.DataField(ref _speedModifier, "speed", 1);
+            serializer.DataField(ref _useSound, "useSound", string.Empty);
+            serializer.DataField(ref _useSoundCollection, "useSoundCollection", string.Empty);
+        }
+
+        /// <summary>
+        ///     Status modifier which determines whether or not we can act as a tool at this time
+        /// </summary>
+        public virtual bool CanUse()
+        {
+            return true;
+        }
+
+        protected void PlaySoundCollection(string name, float volume=-5f)
+        {
+            var soundCollection = _prototypeManager.Index<SoundCollectionPrototype>(name);
+            var file = _robustRandom.Pick(soundCollection.PickFiles);
+            _entitySystemManager.GetEntitySystem<AudioSystem>()
+                .Play(file, Owner, AudioParams.Default.WithVolume(volume));
+        }
+
+        public void PlayUseSound()
+        {
+            if(string.IsNullOrEmpty(UseSoundCollection))
+                _audioSystem.Play(UseSound, Owner);
+            else
+                PlaySoundCollection(UseSoundCollection, 0f);
+        }
+
+        public override ComponentState GetComponentState()
+        {
+            return new ToolComponentState(Behavior);
+        }
+
+        public void AfterAttack(AfterAttackEventArgs eventArgs)
+        {
+            if (Behavior != Tool.Crowbar)
```

Comment:
I *really* don't like this being hardcoded into the base `ToolComponent` like this.

At the same time I also recognize why it is the way it is, hrm.

Reply:
I don't see why it is - could you explain? Is a `CrowbarActAfter` not somehow appropriate?

Reply:
I guess that might work.

Reply:
Hmm, I don't see where I should put this logic. Any ideas?

Reply:
In the end I made this its own component.